src/dinglehopper/align.py

```diff
@@ -15,7 +15,7 @@ def align(t1, t2):
     return seq_align(s1, s2)
 
 
-def score_hint(er: float, n: int) -> Optional[int]:
```

Comment:
Using `Optional` implies that the function might not return anything in some circumstances but here we might return `None`.

Reply:
`Optional[int]` and `Union[int, None]` are equivalent. With Python [3.10+](https://peps.python.org/pep-0604/) you could also write `int|None`